metadata/src/main/java/com/redhat/lightblue/metadata/ValueGenerator.java

```diff
@@ -0,0 +1,42 @@
+package com.redhat.lightblue.metadata;
+
+import java.io.Serializable;
+import java.util.Properties;
+
+/**
+ * Represents value generator definition located in entityInfo.
+ *
+ * @author mpatercz
+ *
+ */
+public class ValueGenerator implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    public enum ValueGeneratorType {
+        IntSequance;
+    }
+
```

Comment:
I don't understand the question.

In entityInfo:

{
    "name":"testSeq",
    "type":"integer"
    ....
}

The configuration part will be handled by the back end anyway.



On Tue, Aug 18, 2015 at 7:20 AM, Marek <notifications@github.com> wrote:

> In
> metadata/src/main/java/com/redhat/lightblue/metadata/ValueGenerator.java
> <https://github.com/lightblue-platform/lightblue-core/pull/431#discussion_r37296267>
> :
>
> > +import java.util.Properties;
> > +
> > +/**
> > + * Represents value generator definition located in entityInfo.
> > + *
> > + * @author mpatercz
> > + *
> > + */
> > +public class ValueGenerator implements Serializable {
> > +
> > +    private static final long serialVersionUID = 1L;
> > +
> > +    public enum ValueGeneratorType {
> > +        IntSequance;
> > +    }
> > +
>
> You mean value generator definition in entityInfo is to be handled by
> value generator type to/from json methods?
>
>            {
>                 "name": "testSeq",
>                 "type": "IntSequance",
>                 "configuration": [
>                     {
>                         "name": "min",
>                         "value": "10000"
>                     },
>                     {
>                         "name": "max",
>                         "value": "99999"
>                     }
>                 ]
>             }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/431/files#r37296267>
> .
>